### PR TITLE
Feat: Open private chat with specific friend

### PR DESCRIPTION
**Pull request by Julieta11**

### CODE REVIEW

File: src/MessagingClient.ts [hunk 1]
```diff
@@ -83,6 +83,22 @@ export class MessagingClient implements MessagingAPI {
...
+    getTotalUnseenMessages(): number {
+        const rooms: Array<any> = this.getAllRooms()
+        return rooms
+            .filter(room => room.getMyMembership() === 'join') // Consider rooms that I have joined
+            .map(room => {
+                return {
+                    unreadMessages: this.getRoomUnreadMessages(room).length,
+                }
+            })
+            .map(unread => unread.unreadMessages)
```
Comment (jmoguilevsky) [inline src/MessagingClient.ts:96]: Why not perform both maps in a single one?

**Review by jmoguilevsky — APPROVED** [review]

File: src/MessagingClient.ts [hunk 2]
```diff
@@ -76,6 +76,20 @@ export class MessagingClient implements MessagingAPI {
         .map((conv): Conversation => conv.conversation)
     }
 
+    /** Get total number of unseen messages from all conversations the user has joined */
+    getTotalUnseenMessages(): number {
+        const rooms = this.getAllRooms()
+        return rooms
+            .filter(room => room.getMyMembership() === 'join') // Consider rooms that I have joined
+            .map(room => {
```
Comment (moliva) [inline src/MessagingClient.ts:84]: Avoid this `map` altogether and use the `getRoomUnreadMessages` in the `reduce` to iterate through elements once here.

File: src/MessagingClient.ts [hunk 3]
```diff
@@ -97,6 +111,19 @@ export class MessagingClient implements MessagingAPI {
         await this.matrixClient.sendReadReceipt(event)
     }
 
+    /** Mark all messages in the conversation as seen */
+    async markMessagesAsSeen(conversationId: ConversationId): Promise<void> {
+        const room = this.matrixClient.getRoom(conversationId)
+        if (!room) {
```
Comment (moliva) [inline src/MessagingClient.ts:117]: Add a comment on why we are returning here clarifying room does not exist, we don't need to do anything.

**Review by moliva — APPROVED** [review]
LGTM!

**Pull request merged**